packages/tab/index.js

```diff
@@ -0,0 +1,192 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import TabIndicator from '@material/react-tab-indicator';
+import {MDCTabFoundation} from '@material/tab/dist/mdc.tab';
+
+export default class Tab extends Component {
+  allowTransitionEnd_ = false;
+  foundation_ = null;
+  tabElement_ = React.createRef();
+  tabContentElement_ = React.createRef();
+  tabIndicatorElement_ = React.createRef();
+
+  state = {
+    'classList': new Set(),
+    'aria-selected': undefined,
+    'tabIndex': undefined,
+  };
+
+  componentDidMount() {
+    this.foundation_ = new MDCTabFoundation(this.adapter);
+    this.foundation_.init();
+
+    if (this.props.active) {
+      this.foundation_.activate();
+    }
+  }
+
+  componentWillUnmount() {
+    this.foundation_.destroy();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.active !== prevProps.active) {
+      if (this.props.active) {
+        this.foundation_.activate(this.props.previousActiveClientRect);
+      } else {
+        this.foundation_.deactivate();
+      }
+    }
+  }
+
+  get classes() {
+    const {classList} = this.state;
+    const {className, minWidth, stacked} = this.props;
+    return classnames('mdc-tab', Array.from(classList), className, {
+      'mdc-tab--min-width': minWidth,
+      'mdc-tab--stacked': stacked,
+    });
+  }
+
+  get adapter() {
+    return {
+      addClass: (className) => {
+        const classList = new Set(this.state.classList);
+        classList.add(className);
+        this.setState({classList});
+      },
+      removeClass: (className) => {
+        const classList = new Set(this.state.classList);
+        classList.delete(className);
+        this.setState({classList});
+      },
+      hasClass: (className) => this.classes.split(' ').includes(className),
+      setAttr: (attr, value) => this.setState({[attr]: value}),
+
+      registerEventHandler: () => this.allowTransitionEnd_ = true,
+      deregisterEventHandler: () => this.allowTransitionEnd_ = false,
+      getOffsetLeft: () => this.tabElement_.current && this.tabElement_.current.offsetLeft,
+      getOffsetWidth: () => this.tabElement_.current && this.tabElement_.current.offsetWidth,
+      getContentOffsetLeft: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetLeft,
+      getContentOffsetWidth: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetWidth,
+      focus: () => this.tabElement_.current && this.tabElement_.current.focus(),
+
+      // activateIndicator, deactivateIndicator, and computeIndicatorClientRect
```

Comment:
I don't think activate/deactivate are necessary, but we can chat about the `computeIndicatorClientRect` method in person

Reply:
Hmmm, I am out tomorrow but I do think we need to talk in person. I say we talk Monday. Maybe you and Pat can try and do a quick audit of the MDCW TabAdapter on Friday, so that you have some detailed technical information. But lets talk Monday and wait to merge until that talk.